pyccel/codegen/printing/ccode.py

```diff
@@ -281,14 +281,82 @@ def _get_comment(self, text):
     def _format_code(self, lines):
         return self.indent_code(lines)
 
-    def _flatten_list(self, irregular_list):
-        if isinstance(irregular_list, (PythonList, PythonTuple)):
-            f_list = [element for item in irregular_list for element in self._flatten_list(item)]
-            return f_list
+    #========================== Numpy Elements ===============================#
+
+    def create_literal_array(self, arg, dvar, name=None):
+        self.add_import(c_imports['ndarrays'])
+        self.add_import(c_imports['string'])
+        shape = str(len(arg))
+        dtype = dvar["dtype"]
+        declare_dtype = dvar["declare_dtype"]
+        shape_dtype = self.find_in_dtype_registry('int', 8)
+        shape_Assign = "("+ shape_dtype +"[]){" + shape + "}"
+        is_view = 'false'
+        temp_literal_array_name = name
+        temp_array_declaration = ""
+        array_create = ""
+        nd = "1"
+        if name is None:
+            order = 'order_c'
+            temp_literal_array_name = self.scope.get_new_name('temp_array')
+            temp_array_declaration = f"t_ndarray {temp_literal_array_name} = " "{.shape = NULL};\n"
+            array_create = f"{temp_literal_array_name} = array_create({nd}, {shape_Assign}, {dtype}, {is_view}, {order});\n"
         else:
-            return [irregular_list]
+            order = dvar['order']
+        dummy_array_name = self.scope.get_new_name('array_dummy')
+        literalList = "{" + ', '.join(self._print(elem) for elem in arg) + "}"
+        dummy_array = f"{declare_dtype} {dummy_array_name}[] = {literalList};\n"
+        cpy_data = f"memcpy({temp_literal_array_name}.{dtype}, {dummy_array_name}, {temp_literal_array_name}.buffer_size);\n"
+
+        return (temp_array_declaration + array_create + dummy_array + cpy_data, temp_literal_array_name)
+
+    def create_variable_array(self, shape, array_names, dvar, name=None):
+        self.add_import(c_imports['ndarrays'])
+        if not name:
+            order = 'order_c'
+            temp_variable_array_name = self.scope.get_new_name("composed")
+        else:
+            order = dvar["order"]
+            temp_variable_array_name = name
+        nd = len(shape)
+        shape = ', '.join(self._print(elem) for elem in shape)
+        is_view = 'false'
+        temp_array_declaration = ""
+        array_create = ""
+        dtype = dvar["dtype"]
+        if name is None:
+            temp_array_declaration = f"t_ndarray {temp_variable_array_name} = " "{.shape = NULL};\n"
+            shape_dtype = self.find_in_dtype_registry('int', 8)
+            shape_Assign = "("+ shape_dtype +"[]){" + shape + "}"
+            array_create = f"{temp_variable_array_name} = array_create({nd}, {shape_Assign}, {dtype}, {is_view}, {order});\n"
+        copy_operations = ""
+        for index, array in enumerate(array_names):
+            offset = f"(({array}.buffer_size) * {index}) / {array}.type_size" if order == "order_c" else index
```

Comment:
You should use `PyccelMul` to create the multiplications here. This will allow you to use the `simplify=True` argument which is recommended for objects created inside pyccel (ie objects which do not have an equivalent in the user's code). This improves readability. E.g. `PyccelMul(a, 0)` will print as `0`

Reply:
Okey, I will create a helper function for it, since PyccelMul takes ```PyccelAstNode``` and the variables there are ```str```
I will also use length instead of that long operation.